WooCommerce/Classes/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardView.swift

```diff
@@ -40,17 +54,59 @@ final class BlazeCampaignDashboardViewHostingController: SelfSizingHostingContro
+    }
+
+    private func createProductSelectorViewController() -> ProductSelectorViewController {
+        let productSelectorViewModel = ProductSelectorViewModel(
+            siteID: viewModel.siteID,
+            purchasableItemsOnly: true,
+            onProductSelectionStateChanged: { [weak self] product in
+                guard let self = self else { return }
+
+                // Navigate to Campaign Creation Form once a simple/bundle product is selected.
+                self.navigateToCampaignCreation(source: .myStoreSection)
+            },
+            onCloseButtonTapped: { [weak self] in
+                guard let self = self else { return }
+
+                blazeNavigationController?.dismiss(animated: true, completion: nil)
+            }
+        )
+
+        return ProductSelectorViewController(configuration: ProductSelectorView.Configuration.configurationForBlaze,
+                                             source: .blaze,
+                                             viewModel: productSelectorViewModel)
+    }
+
+    /// Handles navigation to the campaign creation view
     func navigateToCampaignCreation(source: BlazeSource, productID: Int64? = nil) {
-        let webViewModel = BlazeWebViewModel(siteID: viewModel.siteID,
-                                             source: source,
-                                             siteURL: viewModel.siteURL,
-                                             productID: productID) { [weak self] in
-            self?.handlePostCreation()
+        if ServiceLocator.featureFlagService.isFeatureFlagEnabled(.blazei3NativeCampaignCreation) {
+            let campaignCreationFormViewModel = BlazeCampaignCreationFormViewModel(siteID: viewModel.siteID,
+                                                                                        onCompletion: { })
+            let controller = BlazeCampaignCreationFormHostingController(viewModel: campaignCreationFormViewModel)
+            blazeNavigationController?.show(controller, sender: self)
+        }
+        else {
+            let webViewModel = BlazeWebViewModel(siteID: viewModel.siteID,
+                                                 source: source,
+                                                 siteURL: viewModel.siteURL,
+                                                 productID: productID) { [weak self] in
+                self?.handlePostCreation()
+            }
+            let webViewController = AuthenticatedWebViewController(viewModel: webViewModel)
+            parentNavigationController?.show(webViewController, sender: self)
+            viewModel.didSelectCreateCampaign(source: source)
+        }
+    }
+
+    /// Handles navigation to the Blaze product selector view
+    func navigateToBlazeProductSelector(source: BlazeSource) {
+        if let blazeNavController = blazeNavigationController {
+            blazeNavController.modalPresentationStyle = .pageSheet
```

Comment:
Nit: We usually let the system decide the presentation style (`.automatic`), so we can remove this line.

Reply:
TIL for me, thanks

https://github.com/woocommerce/woocommerce-ios/pull/11625/commits/864f2b31551b83acef03aa79748c637d5d32e244